go-backend/db/organization.go

```diff
@@ -0,0 +1,216 @@
+package db
+
+import (
+	"context"
+	"time"
+	"regexp"
+	logger "github.com/sirupsen/logrus"
+)
+
+const (
+	createOrganizationQuery = `INSERT INTO organizations (
+		name,
+		contact_email,
+		domain_name,
+		subscription_status,
+		subscription_valid_upto,
+		hi5_limit,
+		hi5_quota_renewal_frequency,
+		timezone,
+		created_by,
+		updated_by)
+		VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) RETURNING id`
+
+	updateOrganizationQuery = `UPDATE organizations SET (
+		name,
+		contact_email,
+		domain_name,
+		subscription_status,
+		subscription_valid_upto,
+		hi5_limit,
+		hi5_quota_renewal_frequency,
+		timezone,
+		updated_by,
+		updated_on) = 
+		($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) where id = $11`
+
+	deleteOrganizationQuery = `DELETE FROM organizations WHERE id = $1`
+
+	getOrganizationQuery = `SELECT id,
+		name,
+		contact_email,
+		domain_name,
+		subscription_status,
+		subscription_valid_upto,
+		hi5_limit,
+		hi5_quota_renewal_frequency,
+		timezone,
+		created_by,
+		created_on,
+		updated_by,
+		updated_on FROM organizations WHERE id=$1`
+	
+	listOrganizationsQuery =`SELECT id,
+		name,
+		contact_email,
+		domain_name,
+		subscription_status,
+		subscription_valid_upto,
+		hi5_limit,
+		hi5_quota_renewal_frequency,
+		timezone,
+		created_by,
+		created_on,
+		updated_by,
+		updated_on FROM organizations ORDER BY name ASC`
+	emailRegex = `^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$`
+	domainRegex = `(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]`
+)
+
+type Organization struct {
+	ID                       int				`db:"id" json:"id" `
+	Name                     string				`db:"name" json:"name"`
+	ContactEmail             string				`db:"contact_email" json:"email"`
+	DomainName               string				`db:"domain_name" json:"domain_name"`
+	SubscriptionStatus       int				`db:"subscription_status" json:"subscription_status"`
+	SubscriptionValidUpto    int64				`db:"subscription_valid_upto" json:"subscription_valid_upto"`
+	Hi5Limit                 int				`db:"hi5_limit" json:"hi5_limit"`
+	Hi5QuotaRenewalFrequency string				`db:"hi5_quota_renewal_frequency" json:"hi5_quota_renewal_frequency"`
+	Timezone                 string				`db:"timezone" json:"timezone"`
+	CreatedBy                int				`db:"created_by" json:"_"`
+	CreatedOn                int64				`db:"created_on" json:"_"`
+	UpdatedBy                int				`db:"updated_by" json:"_"`
+	UpdatedOn                int64				`db:"updated_on" json:"_"`
+}
+
+//TODO how to declare this as reusable
+type ErrorResponse struct {
+	Code 	string `json:"code"`
+	Message	string `json:"message"`
+	Fields map[string]string `json:"fields"`
+}
+
+var validEmail = regexp.MustCompile(emailRegex)
```

Comment:
since these are global variables and can be used across the package, their names should be super readable.

Instead of calling it `validEmail` which might be confusing, let's call it `emailRegex`, so that by name we understand that this refers to email regular expression.

same goes for validDomain.